src/Runner.Worker/Handlers/CompositeActionHandler.cs

```diff
@@ -86,13 +83,21 @@ public async Task RunAsync(ActionRunStage stage)
 
                 ExecutionContext.StepTelemetry.HasPreStep = Data.HasPre;
                 ExecutionContext.StepTelemetry.HasPostStep = Data.HasPost;
-                
+
                 ExecutionContext.StepTelemetry.HasRunsStep = hasRunsStep;
                 ExecutionContext.StepTelemetry.HasUsesStep = hasUsesStep;
                 ExecutionContext.StepTelemetry.StepCount = steps.Count;
             }
             ExecutionContext.StepTelemetry.Type = "composite";
 
+            // save job defaults run to be restored after the composite run
+            IDictionary<string, string> jobDefaultsRun = new Dictionary<string, string>();
+            if (ExecutionContext.Global.JobDefaults.ContainsKey("run"))
```

Comment:
Lets not overwrite job defaults, let's use a new key called `CompositeDefaults`, to minimize risk of these changes and make it clearer to read the code where these values are coming from.

Reply:
I changed it but I am not sure if this is what you have meant.